nipoppy/trackers/run_tracker.py

```diff
@@ -43,47 +48,76 @@ def run(global_configs, dash_schema_file, pipelines, session_id="ALL", run_id=1,
-            logger.info(f"n_participants: {n_bids_participants}, session_ids: {session_ids}")
-            logger.info("-"*50)
-
-            status_check_dict = pipe_tracker.get_pipe_tasks(tracker_configs, PIPELINE_STATUS_COLUMNS, pipeline, version)
-
-            dash_col_list = list(schema["GLOBAL_COLUMNS"].keys()) 
-            
-            logger.info(f"Checking session: {session_id}")    
-            _df = pd.DataFrame(index=bids_participants, columns=dash_col_list)          
-            _df["session"] = session
-            _df["pipeline_name"] = pipeline        
+        # Grab BIDS participants from the doughnut
+        doughnut_file = f"{DATASET_ROOT}/scratch/raw_dicom/doughnut.csv"
+        doughnut_df = load_status(doughnut_file)
+        participants_total = doughnut_df[doughnut_df[COL_CONV_STATUS]][COL_BIDS_ID_MANIFEST].unique()
+        n_participants_total = len(participants_total)
+
+        logger.info("-"*50)
+        logger.info(f"pipeline: {pipeline}, version: {version}")
+        logger.info(f"n_participants_total: {n_participants_total}, session_ids: {session_ids}")
+        logger.info("-"*50)
+
+        status_check_dict = pipe_tracker.get_pipe_tasks(tracker_configs, PIPELINE_STATUS_COLUMNS, pipeline, version)
+
+        # only use non-prefixed columns at this stage
+        # for prefixed columns we need to generate the column name
+        dash_col_list = list(key for key, value in schema["GLOBAL_COLUMNS"].items() if not value["IsPrefixedColumn"])
+
+        for session_id in session_ids:
+            session = session_id_to_bids_session(session_id)
+            logger.info(f"Checking session: {session}")
+
+            participants_session = doughnut_df[(doughnut_df[COL_BIDS_ID_MANIFEST].isin(participants_total)) & (doughnut_df[COL_SESSION_MANIFEST] == session)][COL_BIDS_ID_MANIFEST].drop_duplicates().astype(str).str.strip().values
+            n_participants_session = len(participants_session)
+            logger.info(f"n_participants_session: {n_participants_session}")
+
+            _df = pd.DataFrame(index=participants_session, columns=dash_col_list)
+            _df[COL_SESSION_MANIFEST] = session
+            _df["pipeline_name"] = pipeline
             _df["pipeline_version"] = version
+            _df["has_mri_data"] = TRUE # everyone in the status file has MRI data
+
+            tracker_csv = Path(mr_proc_root_dir, 'derivatives', FNAME_BAGEL)
+            if tracker_csv.exists():
+                old_proc_status_df_full = load_bagel(tracker_csv)
+
+                old_proc_status_df_session = old_proc_status_df_full.loc[old_proc_status_df_full[COL_SESSION_MANIFEST] == session]
+                old_participants_session = set(old_proc_status_df_session[COL_BIDS_ID_MANIFEST])
+                old_pipelines_session = set(old_proc_status_df_session['pipeline_name'])
+                
+                # make sure the number of participants is consistent across pipelines
```

Comment:
Not sure if this is before or after our discussion on querying doughnut regardless a full run or selective subset of participant run. But to me it should be possible to ensure number of participants is consistent across pipelines automatically. 
Th runtime error is likely to confuse the user more and trigger slow full runs unnecessarily.

Reply:
This was from before the discussion yesterday. I didn't add the feature where the user can specify specific subjects to run yet because I think it's better to have it in another PR. I think we can address the consistent participant thing in that other PR also? The only thing that that doesn't sit too well with me is that we might have out-of-sync tracking info for different pipelines for the same participants (ex: fMRIPrep is up-to-date but MRIQC is not because for some reason it wasn't run). So that's why I had the error, though I could change it to a warning for now